test/saml-post-signing-tests.spec.ts

```diff
@@ -1,27 +1,129 @@
 import * as fs from "fs";
 import { signSamlPost, signAuthnRequestPost } from "../src/passport-saml/saml-post-signing";
-import { SamlOptions, SamlSigningOptions } from "../src/passport-saml/types";
+import { SamlSigningOptions } from "../src/passport-saml/types";
+import { parseXml2JsFromString } from "../src/passport-saml/xml";
 
 const signingKey = fs.readFileSync(__dirname + "/static/key.pem");
 
 describe("SAML POST Signing", function () {
-  it("should sign a simple saml request", function () {
+  it("should sign a simple saml request", async function () {
     const xml =
       '<SAMLRequest><saml2:Issuer xmlns:saml2="urn:oasis:names:tc:SAML:2.0:assertion">http://example.com</saml2:Issuer></SAMLRequest>';
     const result = signSamlPost(xml, "/SAMLRequest", { privateKey: signingKey });
-    result.should.match(/<DigestValue>[A-Za-z0-9/+=]+<\/DigestValue>/);
-    result.should.match(/<SignatureValue>[A-Za-z0-9/+=]+<\/SignatureValue>/);
+    const doc = await parseXml2JsFromString(result);
+    doc.should.be.deepEqual({
```

Comment:
This is certainly more robust than what we had before. I think this is why a lot of tests compare to a static file: this makes the individual test not fit on a single screen. I can't think of a great solution right now, so I'm not looking for a change, but am open to ideas, perhaps in a follow-on PR.